tests/testsuite/precise_pre_release.rs

```diff
@@ -41,20 +41,84 @@ perhaps a crate was updated and forgotten to be re-vendored?"#,
 }
 
 #[cargo_test]
-fn feature_exists() {
+fn update_pre_release() {
+    cargo_test_support::registry::init();
+
+    for version in ["0.1.1", "0.1.2-pre.0"] {
+        cargo_test_support::registry::Package::new("my-dependency", version).publish();
+    }
+
     let p = project()
         .file(
             "Cargo.toml",
             r#"
             [package]
             name = "package"
+            [dependencies]
+            my-dependency = "0.1.1"
             "#,
         )
         .file("src/lib.rs", "")
         .build();
 
-    p.cargo("-Zprecise-pre-release update")
+    p.cargo("update -p my-dependency --precise 0.1.2-pre.0 -Zunstable-options")
```

Comment:
nit: `-p` is no longer needed.

```suggestion
    p.cargo("update my-dependency --precise 0.1.2-pre.0 -Zunstable-options")
```

Reply:
Sounds gread, no need to pass -p anymore.